attestation-service/verifier/src/lib.rs

```diff
@@ -89,12 +89,28 @@ pub type TeeEvidenceParsedClaim = serde_json::Value;
 
 #[async_trait]
 pub trait Verifier {
-    /// Verify the hardware signature and report data in TEE quote.
-    /// If the verification is successful, a key-value pairs map of TCB status will be returned,
-    /// The policy engine of AS will carry out the verification of TCB status.
+    /// Verify the hardware signature.
+    ///
+    ///
+    /// If `report_data` is given, the binding of the `report_data`
+    /// against the `report_data` inside the hardware evidence will
+    /// be checked. So do `init_data_hash`.
+    ///
+    ///
+    /// Semantically, a `report_data` is a byte slice given when
+    /// a hardware evidence is generated. The `report_data` will be
+    /// included inside the hardware evidence, thus its integrity will
+    /// be protected by the signature of the hardware.
+    ///
+    ///
+    /// A `init_data_hash` is another byte slice given when the TEE
+    /// instance is created. It is always provided by untrusted host,
+    /// but its integrity will be protected by the tee evidence.
+    /// Typical `init_data_hash` is `HOSTDATA` for SNP.
```

Comment:
I wonder if `host_data` would be more clear than `init_data`? I'm not sure what is more common.

Btw this API seems to imply that the client of the verifier might know the content of the init data ahead of time. This is true for the report data, which should always be known to the KBS, but I'm not sure if it's true for the host data. Would it make more sense to think of the host data as a claim that results from verification rather than an input to verification?

Reply:
Overall I think your suggestions are great and maybe I can do this in another PR as many people have approved already, it would make trouble for them to re-review.

My original idea of this is that by changing the `verifier` API like this we can then based on the crate to build a new CoCo-AS, which can work separately without KBS. like this [doc](https://github.com/Xynnn007/kbs/blob/refactor-as-change-api/attestation-service/attestation-service/README.md#api-1). In such case, a user might or might not know the init_data & report_data, though they can decide whether or not to check the binding relationship.

You have raised a great point that we should also wrap the report_data and init_data as claims in the result claim with a more generic name, s.t. extra field `init_data` rather than `mrconfigid` that is too arch specific.

Reply:
About the name. TBO when we discussed the mechanism within the team we always used this terminal, as we thought hostdata is somehow easily understood in a narrow sense as SNP hostdata

Reply:
Ok. Sounds good. We can move forward her and figure out the rest later.